src/ILCompiler.Compiler/src/IL/ILImporter.Scanner.cs

```diff
@@ -275,7 +277,7 @@ private void ImportCall(ILOpcode opcode, int token)
 
             // If we're scanning the fallback body because scanning the real body failed, don't trigger cctor.
             // Accessing the cctor could have been a reason why we failed.
-            if (!_isFallbackBodyCompilation)
+            if (!_isFallbackBodyCompilation && !method.Signature.IsStatic)
```

Comment:
This is trying to capture the difference between when the cctor check is triggered at the callsite vs. at the beginning of the method (when RyuJIT calls `fgInitThisClass`). I haven't actually dug through RyuJIT sources but this made a problem with non-matching generic dictionaries disappear :/.